src/ui/DynamicFacet/DynamicFacet.ts

```diff
@@ -686,6 +683,14 @@ export class DynamicFacet extends Component implements IAutoLayoutAdjustableInsi
     this.queryController.executeQuery(options);
   }
 
+  public async triggerNewIsolatedQuery(beforeExecuteQuery?: () => void) {
+    this.beforeSendingQuery();
+    beforeExecuteQuery && beforeExecuteQuery();
+
+    const results = await this.dynamicFacetQueryController.executeIsolatedQuery();
```

Comment:
Should I do something if I don't get any results? It would be very unlikely...

Reply:
Assuming we catch errors as per the comment above, the return value should at worst case be an empty `IQueryResults` object which we can pass to `handleQuerySuccess`.